lib/uvobot/details/parser.rb

```diff
@@ -0,0 +1,49 @@
+require 'nokogiri'
+require_relative 'procurement_result'
+require_relative 'procurement_announcement'
+
+module Uvobot
+  module Details
+    class Parser
+      class DetailRecognitionError < StandardError
+      end
+
+      def initialize(html)
+        @html_doc = Nokogiri::HTML(html)
+        extend_instance_with(parse_type)
+      end
+
+      def extend_instance_with(module_name)
+        extend Kernel.const_get("Uvobot::Details::#{module_name}")
```

Comment:
Toto je uz kuzelna prasacinka. 

1) Toto nutne nemusi byt dynamicke. A ak chceme, tak sa to da ovela jednoduchsie.
2) Moduly/mixiny su na toto antipattern, normalne dedenie (ak treba)

Cize v prvom kole by som proste nevymyslal a spravil proste metodu `parse_procurement_announcement` a `parse_procurement_result`, ktore dostanu ako parameter `html_doc` a vratia to naformatovane.

Cize ta rozhodovacia logika sa presunie do `UvoParser.parse_detail(html)` a 

```ruby
def self.parse_detail(html)
  case type_header_text
    when 'OZNÁMENIE O VYHLÁSENÍ VEREJNÉHO OBSTARÁVANIA'
      parse_procurement_announcement(html)
    when 'OZNÁMENIE O VÝSLEDKU VEREJNÉHO OBSTARÁVANIA'
parse_procurement_result
 +        else
 +          raise DetailRecognitionError
 +        end
end
```